Source/Utilis/Protos/GenericMessage+Content.swift

```diff
@@ -22,192 +22,102 @@ import Foundation
 
 public extension GenericMessage {
     var hasText: Bool {
-        guard let content = content else { return false }
-        switch content {
-        case .text:
-            return true
-        default:
-            return false
+        guard let _ = messageData as? Text else {
+            let ephemeral = messageData as? Ephemeral
+            return ephemeral?.hasText ?? false
         }
+        return true
     }
     
     var hasConfirmation: Bool {
-        guard let content = content else { return false }
-        switch content {
-        case .confirmation:
-            return true
-        default:
-            return false
-        }
+        return messageData is Confirmation
     }
     
     var hasReaction: Bool {
-        guard let content = content else { return false }
-        switch content {
-        case .reaction:
-            return true
-        default:
-            return false
-        }
+        return messageData is Reaction
     }
     
     var hasAsset: Bool {
-        guard let content = content else { return false }
-        switch content {
-        case .asset:
-            return true
-        default:
-            return false
+        guard let _ = messageData as? WireProtos.Asset else {
+            let ephemeral = messageData as? Ephemeral
+            return ephemeral?.hasAsset ?? false
         }
+        return true
     }
     
     var hasEphemeral: Bool {
-        guard let content = content else { return false }
-        switch content {
-        case .ephemeral:
-            return true
-        default:
-            return false
-        }
+        return messageData is Ephemeral
     }
     
     var hasClientAction: Bool {
-        guard let content = content else { return false }
-        switch content {
-        case .clientAction:
-            return true
-        default:
-            return false
-        }
+        return messageData is ClientAction
     }
     
     var hasCleared: Bool {
-        guard let content = content else { return false }
-        switch content {
-        case .cleared:
-            return true
-        default:
-            return false
-        }
+        return messageData is Cleared
     }
     
     var hasLastRead: Bool {
-        guard let content = content else { return false }
-        switch content {
-        case .lastRead:
-            return true
-        default:
-            return false
-        }
+        return messageData is LastRead
     }
     
     var hasKnock: Bool {
-        guard let content = content else { return false }
-        switch content {
-        case .knock:
-            return true
-        default:
-            return false
+        guard let _ = messageData as? Knock else {
+            let ephemeral = messageData as? Ephemeral
+            return ephemeral?.hasKnock ?? false
         }
+        return true
```

Comment:
Since `messageData` now also unwraps ephemeral content should it be enough to do this:

```suggestion
        return messageData is Knock
```

Reply:
I was thinking that it's not correct to unwrap the ephemeral content. We have a hasEphemeral() method and that's why type of messageData should be as Ephemeral.

Reply:
Well then I don't understand why we unwrap the ephemeral in `hasAsset` and `hasText`. The hasXXX methods behave in consistently (sometimes they include ephemeral content and sometimes they don't), which makes then hard to reason about.

I'd say we should stick to one logic for the hasXXX methods.

Reply:
Done